src/main/java/com/coveros/selenified/element/Element.java

```diff
@@ -544,10 +544,11 @@ public WebElement getWebElement() {
         if (elements.size() > match) {
             return elements.get(match);
         }
-        if (parent != null) {
-            return parent.getWebElement().findElement(defineByElement());
+        String reason = this.prettyOutputStart() + " was not located on the page";
+        if( elements.size() > 0 ) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1155)